Add fire mode and ammo to HMD. Fix weapon text in FMS. Add single fire M229 Hydra

FatLurch opened the pull request:
**When merged this pull request will:**
- Adds single fire mode to M229 Hydras
- Adds fire mode to the HMD per #52 
- Adds ammo count to HMD
- Shortens weapon names on FMS to address #90

FatLurch committed:
Add fire mode #52 and ammo to HMD. Fixes #90 (weapon text in FMS)

FatLurch committed:
Add fire mode #52 and ammo to HMD. Fixes #90 (weapon text in FMS)

YannikH reviewed (approved):
LGTM

ampersand38 commented on addons/uh60_weapons/config/cfgWeapons.hpp:
```diff
@@ -39,10 +39,31 @@ class cfgWeapons {
     magazines[] += {"VTX_PylonRack_M261_M229"};
   };
+  */
+  class missiles_DAR{
```
Need to inherit `class missiles_DAR: RocketPods {`
else you get `No entry 'bin\config.bin/CfgWeapons/missiles_DAR.scope'.`

FatLurch replied:
Addressed in https://github.com/Project-Hatchet/H-60/pull/285/commits/bce1089aa505e96c3e4fdccea86d6442fce648b3

I didn't see this error on my end. Were you seeing this in the dev version or ARMA or is there something else I need to switch on to see this?

Thanks!

BroBeansCPG committed:
Bugfix: Fix EICAS not displaying wheel brake on entry (#186)

* check wheelbrake state on entry

ampersand38 committed:
Fix category of ViV CBA settings (#298)

ampersand38 committed:
fix floor ffv animation (#297)

* fix floor ffv action

* copilot invisible from outside

ampersand38 committed:
Add setting for hook attach script (#296)

YannikH committed:
Feature/sfm flight director (#320)

* Basic config updates for SFM FD
* ALT/RALT/ALTP/HDG/IAS functional in SFM

YannikH committed:
CTAB markers are imported to the JVMF system (#321)

YannikH committed:
Feature/ccfs (#323)

* FMS autopilot only runs on local machine.
* Waypoint selection MP fix
* HDG autopilot corrects for drift
* CCFS fully playable
* psync functional for HDG
* HDG rollover fixed
* FD knob sync fixed
* FMS coupled mode added

YannikH committed:
Feature/mfd improvements (#324)

* Darker MFD backgrounds
* FMS has black background and greener text
* workaround for marking materials
* HMD markings fixed for rockets and hellfires
* Green and yellow colors updated
* TAC iconography updates
* Overhauled vanilla TAC maps
* TAC Scaling fixed
* BFT renders at any scale
* Map overlays for vanilla maps
* added modded maps

Co-authored-by: Lost <34311095+FireStone1029@users.noreply.github.com>

YannikH committed:
Sfm plus (#322)

* SFMPlus

YannikH committed:
0.5.0 Pre-release build (#331)

* Flares fixed
* pylon bug fixed
* torque limits updated to be realistic
* Rotor collision fixed
* Markings application fixed

YannikH committed:
Patch 0.5.1 build

Added APU sound
Fixed APU advisories
Make AFM use SFM torque for damage for now
Fixed IAS and HDG indicators on ND and PFD
Removed CCFS debug text

ZHANGTIANYAO1 committed:
Add simplified Chinese translation (#340)

Add simplified Chinese translation to stringtable.xml

YannikH committed:
Patch/052 (#344)

* turn debug messages off by default

* misc fixes

* HMD/EICAS update

* patch level

* Markings/Lights advisory fix

* Fix EICAS light advisory

* remove 114L

* MFD pages changes

FLIR is now on TAC page instead of ALL
IVHMS is on all in the place of FLIR (besides TAC)
CCFS is now on PFD

* MFD Fixes

Move IVHMS button independent of mfd_any
Add BeansBro to credits
Add IVHMS to ND page

Co-authored-by: BroBeans <cpgbrobeans@gmail.com>
Co-authored-by: BroBeans <BroBeansCPG@users.noreply.github.com>

BradMick committed:
Sfm+ fixes for 0.5.2 (#343)

* Initial Commit

-Adjusted keyboard mode stabilator position.
-Setup stabilator for tuning
-Adjusted input to avoid going over 100% for keyboard users
-Corrected spelling error in masSet functions and adjusted weight to 113.4 (250lbs)

* Update

-Complete retune of the stabilator, tuned for 70, 130 and ~140 knots
-Fixed startup to be correct
-Added a force generator to prevent taking off without the Nr being at 100%
-Tweaked tables for retuned stabilator

* Adjusted force multiplier range

* Table Correction

* Update

-Adjusted getInput to output more consistent values
-Adjusted engCruiseTQTable to use updated input
-Made it so the force generator only generates force when collective is increased

* Update

-Created antiLift script to handle the force generator that prevents the helicopter from taking off.

* Update

Commented out systemChat

* add damage hints

* remove hint in timely fashion

* Adding the engine EH back in to avoid shift-flying

Co-authored-by: BradMick <bradMick@example.com>
Co-authored-by: BroBeans <cpgbrobeans@gmail.com>
Co-authored-by: YannikH <yannikhegge@gmail.com>

YannikH committed:
Feature/qa automation + 0.5.2 engine fixes (#349)

* QA Build creation script
* 0.5.2 Engine handling and SFM+ hotfixes

BroBeansCPG committed:
remove 114L (#351)

ampersand38 committed:
Delete XEH_PREP.hpp.orig (#312)

YannikH committed:
Absolutely massive updates (#360)

* Basic MFD changes

* PFD progress

* EICAS cautions with inverted markers

Functional master caution
Master caution radius increased
PFD ARC display starter
PFD HOVER display starter
PFD ADI overhauled

MFD configuration done to allow disabling of RALT
MFD configuration done to allow RWR damage

* FLIR MFD Failur

* FLIR can break

Flare mempointes fixed
Engine fixes

* CMWS Can fail

* EICAS Caution tapes test

* Initial CAS overlay

* TAC overhaul started

* performance improvements around CAS module

* TAC center rendering code updated

Model geo phys and geo lods updated

* CAS Overlay can be closed

TAC Cursor added (not functional yet)
CAS Overlay render fix

* TAC work

* TAC supports offsets

* TAC Overhaul ready

* getmaptexture lookup fix

* NVG HUD start made

* NVG HUD nearly done

* UH-60 HUD implemented

* Waypoint rendering improved on TAC

ND and ARC broken

* More work on the ND and TAC waypoint rendering, stuff is broken

* Added cursor info

* North align disabled,

Placeholder CAS removed,
Opacity on HMD

* ND Waypoints fixed,

PFD ARC waypoints fixed
TAC hooking now follows

* MFD updated

* Vt7 topo map fixed

AI can fire doorguns now

* Advisories now on overlay

JVMF advisory
LSR advisory
MFD updates

* MFD Multiplayer sync fixed

FLIR multiplayer fixed
FLIR drift fixed
TAC multiplayer use
MFD broken textures added
MFDs auto fix when repaired (broken texture isn't reset)

* FLIR subpage overhaul started

MFD error overhauled

* Hellfire follows PRI CHAN

ALT CHAN implemented
LRFD CHAN implemented
LST MAN and AUTO implemented

* Pylon reassignment implemented

* FLIR boot sequence added

* Basic encoding for MLASS

* Weapons updated

Jettison function added
Pylon reassignment fixed
MH-60M Misc exterior feature added
WIng store damage split
No more anti-lift

* Removed debug messages

YannikH committed:
Loading perf improvements (#363)

* text update p1

* mlass

* - lost's texture fuckery extravaganza

 - MLASS implemented and encoded
 - MH-60M Misc equipment updated

* Hellfire temp fix

LOD4 disappears
EGI/RALT switches encoded

* updated texture

* fuel fix?

* AI Doorguns fixed

Rotor can break again

* Final fixes for XOs misc content

Rotor transparency fixed

* Rotor brake implemented

* rotor brake and RPM limiter functional for AFM

* Fixed the LOD4 disappearing and temp rotor blur fix

* Texture improvements

Cockpit doors can animate
Hellfire model improvements
Overtorque hint fixed
FLIR defaults backwards
All stations can be assigned
All stations can be jetissoned

* Stowed flir visually represented both on model and fullscreen

* Chaingun makes camera shake now

* Rotor spawns at offset

Crew gear improved
Doors can hide again

* UH-60M textures finalized

MH-60M textures finalized
Mass change fixed
Torque limit fixed for overweight
Slingload weight fixed

* Increased HitHrotor hitpoints

Implemented new reflective glass
Cockpit door hiding no longer causes artefacts
Cold start panel functional
RTD controller debug chat removed
Rotor brake debug chat removed
Hellfire and rocket pod models updated
Interactive customization added

* Flight director FMS and HVR modes implemented

Textures improved
Basic cabin light implemented
LST Scan speed increased
Stabilisation code improved
FLIR can destabilise again
FLIR no longer gets stuck on horizon

* Added quick start

Added student driver markings
Fixed EICAS caution marker 40 being on by default

* Removed the sensor panels

Added changelog md

* Loading speed equalized

* reduced laser seek error occurence

reduced amount of conditions used
HVR now usable in AFM
basis for loading screen configd but not implemented

* Altis flooded removed

HH-60G Model and texture added
HH-60G Firefighting texture added
M-230 Sound overhauled

* changelog update

* HH implemented

Old HH skin removed
All skins updated

* MLASS visible again

EICAS properly shows aux tank fuel levels

* updated settings

Damage apply (line 64 gone)
sfm+ hint damge changed to hintsilent
Init settings for hoist changed to be false by default to fix hoist bug (temp)

* fixed simple collective on in sfm+

* Weapon handoff fixed

Co-authored-by: Lost <34311095+FireStone1029@users.noreply.github.com>

mattysmith22 committed:
Build system scons (#392)

* Remove unused pbos

* Add missing header file

* Add inition Scons integration (fails to binarize or pack fully)

* Fix parameter count in preprocessor macros

* Allow optional disabling of binarization for some pbos

* Make scons force prefix using -prefix

Aaren882 committed:
Audio test (#393)

* SFX PBO

* SFX PBO

* SFX PBO

* SFX PBO

* *Add Sound Controllers

* Add Volume Controller Compat

YannikH committed:
Fixed sounds to work with scons (#394)

Added readme for scons

BroBeansCPG committed:
Fix conflict
